ocaml/idl/datamodel.ml

```diff
@@ -6663,17 +6663,44 @@ let vdi_list_changed_blocks = call
     ~allowed_roles:_R_VM_OP
     ()
 
+module Vdi_nbd_server_info = struct
+  let vdi_nbd_server_info =
+    let lifecycle = [Published, rel_inverness, ""] in
+    create_obj
+      ~in_db:false
+      ~persist:PersistNothing
+      ~gen_constructor_destructor:false
+      ~lifecycle
+      ~in_oss_since:None
+      ~name:_vdi_nbd_server_info
+      ~descr:"Details for connecting to a VDI using the Network Block Device protocol"
+      ~gen_events:false
+      ~messages:[]
+      ~doccomments:[]
+      ~messages_default_allowed_roles:(Some []) (* No messages, so no roles allowed to use them *)
+      ~contents:
+      [ (* uid _vdi_nbd_server_info; The uuid is not needed here and only adds inconvenience. *)
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "exportname" "The exportname to request over NBD. This holds details including an authentication token, so it must be protected appropriately. Clients should regard the exportname as an opaque string or token.";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "address" "An address on which the server can be reached; this can be IPv4, IPv6, or a DNS name.";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:Int "port" "The TCP port";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "cert" "The TLS certificate of the server";
+        field ~qualifier:DynamicRO ~lifecycle ~ty:String "subject" "For convenience, this redundant field holds a subject of the certificate.";
```

Comment:
We could discuss adding the host as well to this object. I think it might be useful in case ppl want to back up many VDIs in parallel: it's probably more efficient to download the different VDIs from multiple hosts in parallel. If they have the host reference, then they can do nice optimizations like that. But it's not necessary to add it to this PR, let's get the most basic version in first.

Reply:
Hmm, I though we could have these as `StaticRO` instead, because we just specify these once when we create the object, but it seems that the field of objects with `in_db:false` have `DynamicRO` fields. @robhoes is there a special reason for that, or should we just make all these fields `StaticRO`?

Reply:
We've discussed with Jon that this doesn't matter, because we don't have any generated getters/setters/constructors, so this doesn't affect anything.